### PR TITLE
feat(go/plugins/firebase): add Init method to firebase go plugin

### PR DESCRIPTION
**Pull request by cabljac**
This PR adds an Init method to the Go firebase plugin

Checklist (if applicable):
- [x] Tested (manually, unit tested, etc.)
- [ ] Changelog updated
- [ ] Docs updated

### CODE REVIEW

File: go/plugins/firebase/firebase.go [hunk 1]
```diff
@@ -16,31 +16,69 @@ package firebase
...
+}
+
+// Init initializes the Firebase app with the provided configuration.
+// If called more than once, it logs a message and returns nil.
+func Init(ctx context.Context, cfg *FirebasePluginConfig) error {
+	state.mu.Lock()
+	defer state.mu.Unlock()
+
+	if state.initted {
+		log.Println("firebase.Init: already called, returning without reinitializing")
```
Comment (apascal07) [inline go/plugins/firebase/firebase.go:48]: I wonder if this is the right behavior or if it should error on subsequent calls. Generally this no-op behavior is only deterministic if you are not passing a configuration that you are expecting to take effect. Here, a no-op means you don't know if the initted config matches what you're expecting.

Comment (cabljac) [reply go/plugins/firebase/firebase.go:48]: Good spot, I'll update it :)

File: go/plugins/firebase/firebase.go [hunk 2]
```diff
@@ -16,31 +16,78 @@ package firebase
...
+
+// Init initializes the Firebase app with the provided configuration.
+// If called more than once, it logs a message and returns nil.
+func Init(ctx context.Context, cfg *FirebasePluginConfig) error {
+	state.mu.Lock()
+	defer state.mu.Unlock()
+
+	if state.initted {
+		// throw an error if the app is already initialized
+		return fmt.Errorf("firebase.Init: Firebase app already initialized")
```
Comment (apascal07) [inline go/plugins/firebase/firebase.go:48]:
```suggestion
		return errors.New("firebase.Init: Firebase app already initialized")
```

File: go/plugins/firebase/firebase.go [hunk 3]
```diff
@@ -16,31 +16,78 @@ package firebase
...
-		app = newApp
+	state.app = app
+	state.initted = true
+
+	return nil
+}
+
+// UnInit uninitializes the Firebase app.
+
+func UnInit() {
```
Comment (apascal07) [inline go/plugins/firebase/firebase.go:74]: Test is in the same package, better to drop this and just clean up state directly. We don't want to expose a function like this.

**Review by apascal07 — CHANGES_REQUESTED** [review]

**Comment by cabljac** [comment]
closing in favor of https://github.com/firebase/genkit/pull/791

**Pull request closed without merging**